jobs/standalone-6/templates/config/redis.conf

```diff
@@ -48,4 +48,6 @@ tls-key-file /var/vcap/jobs/standalone-6/config/tls/redis.key
 tls-ca-cert-file /var/vcap/jobs/standalone-6/config/tls/redis.ca
 tls-auth-clients no
 tls-protocols "TLSv1.2 TLSv1.3"
+<% else %>
+port 6379
```

Comment:
How does this interact with the port: 0 in the credential files above?

Reply:
This one configures redis whereas credentials are the ones feeding the information back to blacksmith. Unlike the previous logic which gave `port: 0` when it shouldn't (fixed,) this one results in the following cases:

When **redis.tls.enabled** is _true_ and **redis.tls.dual-mode** is _false_:

```
port 0
tls-port 6379
tls-cert-file /var/vcap/jobs/standalone-6/config/tls/redis.cert
tls-key-file /var/vcap/jobs/standalone-6/config/tls/redis.key
tls-ca-cert-file /var/vcap/jobs/standalone-6/config/tls/redis.ca
tls-auth-clients no
tls-protocols "TLSv1.2 TLSv1.3"
```

When **redis.tls.enabled** is _true_ and **redis.tls.dual-mode** is _true_:

```
port 6379
tls-port 16379
tls-cert-file /var/vcap/jobs/standalone-6/config/tls/redis.cert
tls-key-file /var/vcap/jobs/standalone-6/config/tls/redis.key
tls-ca-cert-file /var/vcap/jobs/standalone-6/config/tls/redis.ca
tls-auth-clients no
tls-protocols "TLSv1.2 TLSv1.3"
```

and when **redis.tls.enabled** is _false_ (no TLS enabled):

```
port 6379
```

With the updated logic on credentials this is identical and when `port 0` is used, it disables non tls communication and credentials report that back to blacksmith, essentially notifying it of the disabled state. I hope that clarifies your question. Let me know if there is a specific part of the logic you'd like me to look further into.